lib/org/chromium/webidl/IDLFragmentExtractor.js

```diff
@@ -21,96 +22,179 @@ foam.CLASS({
       },
     },
     {
+      class: 'Array',
+      of: 'String',
+      documentation: 'Array of IDL Fragments found in the parsed HTML file.',
       name: 'idlFragments',
-      // description: 'all idlFragments found in the parsed HTML file',
+    },
+    {
+      name: 'tags_',
+      documentation: 'Array of tags extracted from file to be processed.',
+      expression: function(file) {
+        var results = this.lexer_.parseString(file.contents).value;
+        if (!results)
+          throw new Error("IDLFragmentExtractor: IDL Parse was not successful.");
+        return results;
+      },
+    },
+    {
+      name: 'lexer_',
+      factory: function() { return this.HTMLLexer.create(); },
+    },
+    {
+      name: 'tagStack_',
+      documentation: 'Stack used to perform HTML Tag matching.',
+      factory: function() {
+        return this.TagStack.create({ isExcluded: this.isExcluded });
+      },
+    },
+    {
+      name: 'tagMatching_',
+      documentation: 'Used to determine if tag matching is currently performed.',
+      value: true,
     },
   ],
 
   methods: [
-    function extract() {
-      var lexer = this.HTMLLexer.create();
-      var OPEN = this.TagType.OPEN.name;
-      var CLOSE = this.TagType.CLOSE.name;
-      var extractAttr = function(node, attrName) {
+    {
+      name: 'extractAttr',
+      documentation: 'Extracts given attribute from tag.',
+      code: function(tag, attrName) {
         var retVal = [];
-        node.attributes.forEach(function(attr) {
+        tag.attributes.forEach(function(attr) {
           if (attr.name === attrName) {
             retVal = retVal.concat(attr.value.split(' '));
           }
         });
         return retVal;
-      };
-
-      var results = lexer.parseString(this.file.contents).value;
-      if (!results) throw new Error("IDL Parse was not successful.");
-
-      var idlFragments = [];
-      var tagStack = [];            // Used for tag matching.
-      var excludeStack = [];        // Used for tracking excluded (example/note) tags.
-      var tagMatching = true;       // Set whe not inside a pre tag of interest.
-      var content = '';             // Used to group together related content.
-      for (var i = 0; i < results.length; i++) {
-        var item = results[i];
-        var isTag = lexer.Tag.isInstance(item);
-
-        // FUTURE: Handle nested pre tags.
-        // As of this writing, there has not been any IDL fragments
-        // that has been found within nested pre tags.
-        if (!tagMatching) {
-          // Ignoring all tags. Only extracting text content within pre tags.
-          if (isTag) {
-            if (item.nodeName === 'pre') {
-              tagMatching = true;
-              tagStack.pop();
-              idlFragments.push(lexer.lib.unescapeString(content));
-              content = '';
+      },
+    },
+    {
+      name: 'isExcluded',
+      documentation: 'Determines if tag has excluded classes.',
```

Comment:
What are "excluded classes", maybe something like ...`has class names that would exclude it and its descendants from IDL extraction`.